src/gui/dialogs/editor/edit_unit.cpp

```diff
@@ -701,6 +700,7 @@ void editor_edit_unit::update_attacks() {
 		find_widget<toggle_button>(get_window(), "range_melee", false).set_value(true);
 		find_widget<toggle_button>(get_window(), "range_ranged", false).set_value(false);
 	}
+
 	if (attack["range"] == "ranged") {
 		find_widget<toggle_button>(get_window(), "range_melee", false).set_value(false);
 		find_widget<toggle_button>(get_window(), "range_ranged", false).set_value(true);
```

Comment:
If the group is a class member now, you can use its API to set the active value.